src/pint/scripts/pint_parameter.py

```diff
@@ -0,0 +1,221 @@
+        "The current output is"
+        "['name', 'description',"
+        " 'unit', 'value', 'aliases']"
+    )
+    parser.add_argument(
+        "-p",
+        "--parameter",
+        nargs="*",
+        default=["all"],
+        help="Quarried parameter names.",
+    )
+    parser.add_argument(
+        "-c",
+        "--component",
+        nargs="*",
+        default=None,
+        help="Print parameters from components.",
+    )
+    parser.add_argument(
+        "-i",
+        "--info",
+        nargs="*",
+        default=["value", "description", "units", "aliases"],
+        help="The requested information from parameter.",
+    )
+    parser.add_argument(
+        "-f",
+        "--format",
+        type=str,
+        default="stdout",
+        help="Output format:\n"
+        "'rst': Sphinx .rst file.\n"
+        "'json': JSON dictionary file.\n "
+        "'stdout': Stand out.",
+    )
+    parser.add_argument(
+        "-o", "--output", type=str, default=None, help="Output file path."
+    )
+
+    args = parser.parse_args()
+
+    # Get the all the components map from all lower case name
+    #  to real component name.
+    all_components = Component.component_types
+    component_name_map = {}
+    for c in all_components.keys():
+        component_name_map[c.lower()] = c
+
+    # Set up the parameter name space including the aliases
+    param_name_map, all_params, prefixed_param = get_param_name_map(all_components)
+
+    # Check the parameter input.
+    if "all" in args.parameter:
+        is_all_param = True
+    else:
+        is_all_param = False
+
+    # Check the component input.
+    if args.component is None:
+        parse_comp = False
+    else:
+        if not is_all_param:
+            raise ValueError(
+                "Currently only support getting all parameters"
+                " from requested components."
+            )
+        else:
+            parse_comp = True
+            if "all" in args.component:
+                quarry_components = list(all_components.keys())
+            else:
+                quarry_components = args.component
+
+    # Get information for all buiting parameters. The result will be a subset
+    # of the this result.
+
+    # When only parse the parameters.
+    if not parse_comp:
+        if is_all_param:
+            quarry_params = all_params
+        else:
+            quarry_params = args.parameter
+    else:
+        quarry_params = []
+        for qc in quarry_components:
+            if qc.lower() not in component_name_map.keys():
+                raise ValueError(
+                    "Component `{}` is not recognised by" " PINT.".format(qc)
+                )
+            else:
+                cp = component_name_map[qc]
+                cp_obj = all_components[cp]()
+                quarry_params += cp_obj.params
+
+    request_info = args.info
+    result = {}
+    # check quarry parameters.
+    if not parse_comp:
+        for q_param in quarry_params:
```

Comment:
I think this should be "query" rather than "quarry"?

Reply:
You are right,